corehq/apps/reports/management/commands/project_stats_report.py

```diff
@@ -295,20 +361,15 @@ def _attachment_sizes(self):
                 ORDER BY content_type, bucket
             """, [self.domain])
 
-            self.stdout.write('Form attachment sizes (bytes)')
-            self._print_table(
-                ['Content Type', 'Count', 'Bucket range', 'Bucket (1-10)'],
-                [
-                    (row.content_type, row.freq, '[%s-%s]' % (row.bucket_min, row.bucket_max), row.bucket)
-                    for row in fetchall_as_namedtuple(cursor)
-                ]
-            )
+            result = [i for i in fetchall_as_namedtuple(cursor)]
+            ResourceModel.set_stat(ResourceModel.ATTACHMENTS, result)
 
     def _ucr(self):
         static_datasources = StaticDataSourceConfiguration.by_domain(self.domain)
         dynamic_datasources = DataSourceConfiguration.by_domain(self.domain)
-        self._print_value('Static UCR data sources', len(static_datasources))
-        self._print_value('Dynamic UCR data sources', len(dynamic_datasources))
+
+        ResourceModel.set_stat(ResourceModel.STATIC_DATASOURCES, len(static_datasources))
+        ResourceModel.set_stat(ResourceModel.DYNAMIC_DATASOURCES, len(dynamic_datasources))
```

Comment:
I know you didn't write this, but this will pull all the configs into memory. You could instead use the reduced couch query with something like: 
```py
        DataSourceConfiguration.view(
            'userreports/data_sources_by_build_info',
            startkey=key,
            endkey=key + [{}],
            reduce=True,
            include_docs=False
        )
```
since the reduce function for that view is `_count`.

Reply:
That could work, yes, but we are not interested only in the count. The actual data is used also [here](https://github.com/dimagi/commcare-hq/blob/c7c5ea9809f2af0808dbbe350e9ebda3875803be/corehq/apps/reports/management/commands/project_stats_report.py#L400).